composer/datasets/cifar.py

```diff
@@ -0,0 +1,252 @@
+                    transforms.RandomHorizontalFlip(),
+                    transforms.ToTensor(),
+                    transforms.Normalize(cifar10_mean, cifar10_std),
+                ])
+            else:
+                transformation = transforms.Compose([
+                    transforms.ToTensor(),
+                    transforms.Normalize(cifar10_mean, cifar10_std),
+                ])
+
+            dataset = CIFAR10(
+                self.datadir,
+                train=self.is_train,
+                download=self.download,
+                transform=transformation,
+            )
+        sampler = dist.get_sampler(dataset, drop_last=self.drop_last, shuffle=self.shuffle)
+
+        return dataloader_hparams.initialize_object(dataset,
+                                                    batch_size=batch_size,
+                                                    sampler=sampler,
+                                                    drop_last=self.drop_last)
+
+
+@dataclass
+class WebCIFAR10DatasetHparams(WebDatasetHparams, SyntheticHparamsMixin):
+    """Defines an instance of the CIFAR-10 WebDataset for image classification.
+
+    Parameters:
+        train_shards (int): Number of training split shards.
+        val_shards (int): Number of validation split shards.
+    """
+    train_shards: int = hp.optional('Training split shards', default=16)
+    val_shards: int = hp.optional('Validation split shards', default=8)
+
+    def initialize_object(self, batch_size: int, dataloader_hparams: DataloaderHparams) -> DataLoader:
+        cifar10_mean = 0.4914, 0.4822, 0.4465
+        cifar10_std = 0.247, 0.243, 0.261
+
+        if self.is_train:
+            split = 'train'
+            size = 50_000
+            n_shards = self.train_shards
+        else:
+            split = 'val'
+            size = 10_000
+            n_shards = self.val_shards
+        size = size - size % n_shards
+
+        if self.use_synthetic:
+            dataset = SyntheticBatchPairDataset(
+                total_dataset_size=size,
+                data_shape=[3, 32, 32],
+                num_classes=10,
+                num_unique_samples_to_create=self.synthetic_num_unique_samples,
+                device=self.synthetic_device,
+                memory_format=self.synthetic_memory_format,
+            )
+        else:
+            if self.is_train:
+                transform = transforms.Compose([
+                    transforms.RandomCrop(32, padding=4),
+                    transforms.RandomHorizontalFlip(),
+                    transforms.ToTensor(),
+                    transforms.Normalize(cifar10_mean, cifar10_std),
+                ])
+            else:
+                transform = transforms.Compose([
+                    transforms.ToTensor(),
+                    transforms.Normalize(cifar10_mean, cifar10_std),
+                ])
+
+            urls = ['/datasets/web_cifar10/%s_%05d.tar' % (split, i) for i in range(n_shards)]
```

Comment:
I'm assuming this will be updated to accept an S3 bucket rather than local paths?